src/dataExplorer/context/persistance.tsx

```diff
@@ -4,40 +4,48 @@ import {DEFAULT_TIME_RANGE} from 'src/shared/constants/timeRanges'
 import {useSessionStorage} from 'src/dataExplorer/shared/utils'
 import {Bucket} from 'src/types'
 
-interface MeasurementSelection {
+export interface SchemaSelection {
   bucket: Bucket
   measurement: string
+  synced: boolean
+  composition?: {
+    startLine: number
+    endLine: number
+  }
 }
 
 interface ContextType {
   horizontal: number[]
   vertical: number[]
   range: TimeRange
   query: string
-  selection: MeasurementSelection
+  selection: SchemaSelection
 
   setHorizontal: (val: number[]) => void
   setVertical: (val: number[]) => void
   setRange: (val: TimeRange) => void
   setQuery: (val: string) => void
-  setSelection: (val: MeasurementSelection) => void
+  setSelection: (val: SchemaSelection) => void
 }
 
+export const DEFAULT_SCHEMA = Object.freeze({
```

Comment:
Object.freeze() means that it cannot be updated. So no one accidentally passing in the same pointer object to be used again and again. Instead, must clone into a mutable object.